Controller/Payconiq/Process.php

```diff
@@ -164,4 +176,15 @@ protected function getList()
 
         return $list;
     }
+
+    public function secondChance($order){
```

Comment:
this should be part of a second chance service class. Do not use/add additional methods, classes and logic in every place a second chance record needs to be created.

Reply:
and the same service class should be used here - https://github.com/buckaroo-it/Magento2/blob/760ded2df4800cb473c4e531b361edc70be618f5/Controller/Redirect/Process.php#L717 (which is almost identical to this one, except in there order is retrieved from the class, here it's passed as argument)

Reply:
fixed done